sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -248,7 +248,11 @@ case class ScalarSubquery(
     children: Seq[Expression] = Seq.empty,
     exprId: ExprId = NamedExpression.newExprId)
   extends SubqueryExpression(plan, children, exprId) with Unevaluable {
-  override def dataType: DataType = plan.schema.fields.head.dataType
+  override def dataType: DataType = if (!plan.schema.fields.isEmpty) {
+    plan.schema.fields.head.dataType
+  } else {
+    NullType
```

Comment:
Let's also add a comment that it returns `NullType` for testing purpose.

Reply:
Added